OpenTerm/Util/Execution/CommandExecutor.swift

```diff
@@ -46,6 +46,8 @@ class CommandExecutor {
     /// Context from commands run by this executor
     private var context = CommandExecutionContext()
 
+    private var commandExecutedCallback: ((ReturnCode) -> Void)?
```

Comment:
IMO, we should either use delegate pattern or callback pattern, not both.

#71 moves the "command finished" delegate method into the `Parser` class as `parserDidEndTransmission`, so this will conflict.

This was done because the command will return a status code, but stdout will most likely come into the pipe afterwards, then is parsed on a different thread, and finally output will be printed to the text view. Only after that should a new prompt be drawn, so the `parserDidEndTransmission` is called when the `Parser` finishes a command, which is after the command exits.

If you only care about the exit code (as you currently do), then adding a callback here would be fine, as it would be called at a different time than the `Parser` delegate `parserDidEndTransmission`, but if you want to capture output, more work will be required, and putting a callback here would not be smart.